src/Wallet.purs

```diff
@@ -187,13 +269,22 @@ apiVersion :: Wallet -> Effect String
 apiVersion = _Wallet.apiVersion
 
 -- | Manually tested and works with Nami.
-enable :: Wallet -> Aff Api
-enable = Promise.toAffE <<< _Wallet.enable
+enable_ :: Warn (Text "enable_ is deprecated, use enable instead") => Wallet -> Aff Api
+enable_ = Promise.toAffE <<< _Wallet.enable
+
+-- | Manually tested and works with Nami.
+enable :: forall r. Wallet -> Aff (Either (Variant (| ApiError + ApiForeignErrors + UnknownError r)) Api)
+enable = toAffEitherE (onCatch <<< rejectionToForeign) <<< _Wallet.enable
+  where
+  onCatch :: Foreign -> Variant (| ApiError + ApiForeignErrors + UnknownError r)
+  onCatch rejection =
+    either foreignErrors (fromMaybe' (\_ -> unknownError rejection) <<< toApiError) $ readWalletError rejection
```

Comment:
@paluh Here's the current solution. Error handling for the other procedures will be filled out similarly.